src/test/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyResponseImplTest.java

```diff
@@ -0,0 +1,107 @@
+import com.microsoft.azure.proton.transport.proxy.ProxyResponse;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.microsoft.azure.proton.transport.proxy.impl.StringUtils.NEW_LINE;
+
+public class ProxyResponseImplTest {
+    /**
+     * Verifies that it successfully parses a valid HTTP response.
+     */
+    @Test
+    public void validResponse() {
+        // Arrange
+        final String[] statusLine = new String[]{"HTTP/1.1", "200", "Connection Established"};
+        final Map<String, List<String>> headers = new HashMap<>();
+        headers.put("FiddlerGateway", Collections.singletonList("Direct"));
+        headers.put("StartTime", Collections.singletonList("13:08:21.574"));
+        headers.put("Connection", Collections.singletonList("close"));
+
+        final String response = TestUtils.createProxyResponse(statusLine, headers);
+        final ByteBuffer contents = TestUtils.ENCODING.encode(response);
+
+        // Act
+        final ProxyResponse actual = ProxyResponseImpl.create(contents);
+
+        // Assert
+        Assert.assertNotNull(actual);
+
+        final HttpStatusLine status = actual.getStatus();
+        Assert.assertEquals("1.1", status.getProtocolVersion());
+        Assert.assertEquals(statusLine[2], status.getReason());
+        Assert.assertEquals(Integer.parseInt(statusLine[1]), status.getStatusCode());
+
+        Assert.assertFalse(actual.isMissingContent());
+
+        final Map<String, List<String>> actualHeaders = actual.getHeaders();
+        Assert.assertEquals(headers.size(), actualHeaders.size());
+
+        headers.forEach((key, value) -> {
+            final List<String> actualValue = actualHeaders.get(key);
+
+            Assert.assertTrue(actualHeaders.containsKey(key));
+            Assert.assertNotNull(actualValue);
+            Assert.assertEquals(1, actualValue.size());
+            Assert.assertEquals(value.get(0), actualValue.get(0));
+        });
+    }
+
+    /**
+     * Verifies that an exception is thrown when the header is invalid. successfully parses a valid HTTP response.
+     */
+    @Test
+    public void invalidHeader() {
+        // Arrange
+        final String[] statusLine = new String[]{"HTTP/1.1", "abc", "Connection Established"};
+        final Map<String, List<String>> headers = new HashMap<>();
+        headers.put("FiddlerGateway", Collections.singletonList("Direct"));
+        headers.put("StartTime", Collections.singletonList("13:08:21.574"));
+        headers.put("Connection", Collections.singletonList("close"));
+
+        final String response = TestUtils.createProxyResponse(statusLine, headers);
+        final ByteBuffer contents = TestUtils.ENCODING.encode(response);
+
+        // Act & Assert
+        try {
+            ProxyResponseImpl.create(contents);
```

Comment:
You can use assertThrows instead of this try-catch block. Or add an assertFail() below this line in the case that no exception is thrown.

Reply:
yes, it is a good suggestion. Updated.